lib/OnyxUtils.ts

```diff
@@ -781,8 +783,7 @@ function sendDataToConnection<TKey extends OnyxKey>(mapping: Mapping<TKey>, val:
         return;
     }
 
-    const valuesWithoutNestedNulls = utils.removeNestedNullValues(val);
-    (mapping as DefaultConnectOptions<TKey>).callback?.(valuesWithoutNestedNulls, matchedKey as TKey);
+    (mapping as DefaultConnectOptions<TKey>).callback?.(value === null ? undefined : value, matchedKey as TKey);
```

Comment:
Can we add a comment why we are returning undefined and not null? This might be hard to put into a shortish comment but it would be very helpful I think